File: Volume 10 (1000 - 1099)/1094/1094.c
#include <stdio.h>
 
int main() {
    int i, n, k, c = 0, r = 0, s = 0, total;
    char ch;
    scanf("%d", &n);
    for (i = 0; i < n; i++) {
        scanf("%d %c", &k, &ch);
        if (ch == 'C') c += k;
        else if (ch == 'S') s += k;
        else if (ch == 'R') r += k;
    }
    total = c+r+s;
    printf("Total: %d cobaias\n", total);
    printf("Total de coelhos: %d\n", c);
    printf("Total de ratos: %d\n", r);
    printf("Total de sapos: %d\n", s);
    printf("Percentual de coelhos: %.2f %%\n", (float)(c*100)/total);
    printf("Percentual de ratos: %.2f %%\n", (float)(r*100)/total);
    printf("Percentual de sapos: %.2f %%\n", (float)(s*100)/total);
    return 0;
}

File: Volume 10 (1000 - 1099)/1094/1094.rb
#!/usr/bin/ruby

c = r = s = 0
for i in 0...gets.to_i
    t = Array.new(gets.split(' '))
    if t[1] == 'C'
        c += t[0].to_f
    elsif t[1] == 'S'
        s += t[0].to_f
    elsif t[1] == 'R'
        r += t[0].to_f
    end
end
total = c+r+s
puts "Total: %d cobaias" % [total]
puts "Total de coelhos: %d" % [c]
puts "Total de ratos: %d" % [r]
puts "Total de sapos: %d" % [s]
puts "Percentual de coelhos: %.2f" % [(c*100)/total] + " %"
puts "Percentual de ratos: %.2f" % [(r*100)/total] + " %"
puts "Percentual de sapos: %.2f" % [(s*100)/total] + " %"